app/src/main/java/com/android/solvit/MainActivity.kt

```diff
@@ -203,66 +207,29 @@ fun SeekerUI(
   val navigationActions = NavigationActions(navController)
   val user by authViewModel.user.collectAsState()
 
-  NavHost(navController = navController, startDestination = Route.SERVICES) {
-    composable(Route.SERVICES) { ServicesScreen(navigationActions, listProviderViewModel) }
-    composable(Route.PROVIDERS) {
-      user?.let { it1 ->
-        SelectProviderScreen(
-            listProviderViewModel = listProviderViewModel,
-            navigationActions = navigationActions,
-            userId = it1.uid,
-            locationViewModel = locationViewModel)
-      }
+  NavHost(navController = navController, startDestination = Route.SEEKER_OVERVIEW) {
+    // Overview
+    composable(Route.SEEKER_OVERVIEW) { ServicesScreen(navigationActions, listProviderViewModel) }
+
+    // Providers
+    composable(Route.PROVIDERS_LIST) {
+      SelectProviderScreen(
+          seekerProfileViewModel,
+          listProviderViewModel,
+          user!!.uid,
```

Comment:
❗ Unsafe use of !! operator with user.uid 🔧 Suggestion: Add proper null safety:

```suggestion
composable(Route.PROVIDERS_LIST) {
    user?.let { nonNullUser ->
        SelectProviderScreen(
            seekerProfileViewModel,
            listProviderViewModel,
            nonNullUser.uid,
            navigationActions,
            locationViewModel)
    } ?: run {
        // Handle null user case, perhaps navigate to login
        navigationActions.navigateTo(Route.AUTH)
    }
}
```

Reply:
If we enter this screen the user is necessarily non-null, the check is done in `SolvItApp`